content/en/security_platform/application_security/troubleshooting.md

```diff
@@ -0,0 +1,444 @@
+  tag: "Documentation"
+  text: "Programming Language and Framework Compatibility"
+---
+
+If you experience unexpected behavior with Datadog Application Security, there are common issues you can investigate, as mentioned below. If you continue to have trouble, reach out to [Datadog support][1] for further assistance. 
+
+## Application Security data is not appearing in Datadog
+
+Check your [tracer startup logs][2]. If `appsec_enabled` is `true`, then Application Security is enabled and running.
+
+Alternatively, you can use the metric `datadog.apm.appsec_host` to check if Application Security is running.
+
+1. Go to **Metrics > Summary** in Datadog.
+2. Search for the metric `datadog.apm.appsec_host`. If the metric doesn’t exist, then there are no services running Application Security. If the metric exists, the services are reported with the metric tags `host` and `service`.
+3. Select the metric, and in the **Tags** section, search for `service` to see which services are running Application Security.
+
+Check for `datadog.trace_agent.appsec.request` metrics to confirm the Datadog Agent is receiving Application Security events.
+
+1. Go to **Metrics > Summary** in Datadog.
+2. Search for `datadog.trace_agent.appsec.request` metrics. If the metrics do not exist, then the Agent is not receiving Application Security events. If the metrics are found, but the Application Security dashboard doesn't show any data, check `datadog.trace_agent.appsec.request_error` to see if there are any errors related to sending the data to the Agent.
+
+If you are not seeing `datadog.apm.appsec_host` or `datadog.trace_agent.appsec.request`, check the [in-app instructions][3] to confirm that all steps for the initial setup are complete.
+
+## Application Security rate limits
+
+Application Security traces are rate-limited to 100 traces per second. Traces sent after the limit are not reported. Contact [Datadog support][1] if you need to change the limit.
+
+## Application Security events are missing in the dashboard
+
+There are a series of steps that must run successfully for Application Security events to appear in the Application Security dashboard. It is important to check each step when investigating this issue. Additional troubleshooting steps for specific languages are in the language tab at the end.
+
+### Send a test attack to your application
+
+Simulate an attack to test your Application Security setup using the following command:
+
+```
+curl -A Arachni/v YOUR_APP_URL
+```
+
+View the resulting Application Security event in the [Application Security dashboard][4]. 
+
+It is possible that the test attack doesn't reach your application because of a firewall or another type of proxy server. It’s known that the following services block the Arachni test:
+
+- AWS API Gateway: Removes the user-agent and malicious parameters, even when its built-in WAF is deactivated
+- Extensible Service Proxy (ESP) by Google 
```

Comment:
This can go away thank to the "dd-scanner-test-log"

Reply:
Removed.